### PR TITLE
samples: nrf9160: download: fix broken link and deprecated mbedtls function names

### PR DESCRIPTION
**Pull request by anhmolt**
Update default https url and certificate to download test file from
nrfconnectsdk aws. The previous https-url is now unreachable
because the github-user that was hosting the page no longer exist.
Test file is uploaded to aws.

Update deprecated mbedtls_256 function names so that the sample
compiles with `CONFIG_SAMPLE_COMPUTE_HASH` and `CONFIG_SAMPLE_COMPARE_HASH`.

NCSDK-14267

### CODE REVIEW

File: samples/nrf9160/download/src/main.c [hunk 1]
```diff
@@ -121,7 +121,7 @@ static int callback(const struct download_client_evt *event)
 		}
 
 #if CONFIG_SAMPLE_COMPUTE_HASH
```
Comment (rlubos) [inline samples/nrf9160/download/src/main.c:123]: Maybe add `CONFIG_SAMPLE_COMPUTE_HASH` to extra configs in sample.yaml, to get a better build coverage?

Comment (anhmolt) [reply samples/nrf9160/download/src/main.c:123]: Sounds good! Added it now.

**Review by rlubos — APPROVED** [review]

File: samples/nrf9160/download/src/main.c (hunk 1 above)
Comment (rlubos) [reply samples/nrf9160/download/src/main.c:123]: Thanks!

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -166,6 +166,10 @@ nrf9160 samples
     * Enabled building of bootloader FOTA update files.
     * Corrected handling of the bootloader FOTA updates.
 
+  * :ref:`download_sample` sample:
+
+    * Updated default https url and certificate due to the old link being broken.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:171]:
```suggestion
    * Updated default HTTPS URL and certificate due to the old link being broken.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -170,6 +170,10 @@ nrf9160 samples
 
     * Updated the library to store credentials and server settings permanently on bootstrap.
 
+  * :ref:`download_sample` sample:
+
+    * Updated default HTTPS URL and certificate due to the old link being broken.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:175]:
```suggestion
    * Updated the default HTTPS URL and certificate due to the old link being broken.
```

**Review by umapraseeda — APPROVED** [review]

**Review by lemrey — APPROVED** [review]
Looks good

**Pull request merged**